d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -211,8 +219,102 @@ public void onSuccess(None result)
         }
         runNextMarkUpOrMarkDown();
       }
-    });
+    };
+
+    final Callback<None> doMarkUpCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e)
+      {
+        // this would ideally never happen
+        callback.onError(e);
+      }
+
+      @Override
+      public void onSuccess(None result)
+      {
+        _log.info("doing mark up of main cluster " + _cluster);
+        _server.markUp(_cluster, _uri, _partitionDataMap, _uriSpecificProperties, markUpCallback);
+      }
+    };
+
+    final Callback<None> warmupMarkDownCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e) {
+        _log.info("mark down warm up cluster " + _warmupClusterName + " has failed");
+        // what to do here ? - go ahead continue to announce main cluster
+        // doAnnounceCallback.onSuccess(None.none());
+      }
+
+      @Override
+      public void onSuccess(None result) {
+        _log.info("mark down warm up cluster " + _warmupClusterName +
+            " has completed, now mark up of main cluster " + _cluster);
+        doMarkUpCallback.onSuccess(None.none());
+      }
+    };
+
+    final Callback<None> doWarmupMarkDownCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e) {
+        _log.info("failure in scheduling warm up mark down");
+        // what to do here ? - go ahead continue to announce main cluster
+        // doAnnounceCallback.onSuccess(None.none());
+      }
+
+      @Override
+      public void onSuccess(None result) {
+        _log.info("warm up for cluster " + _warmupClusterName + " completed, now calling mark down on the warm up cluster");
+        _server.markDown(_warmupClusterName, _uri, warmupMarkDownCallback);
+      }
+    };
+
+    final Callback<None> doWarmupWaitCallback = new Callback<None>() {
```

Comment:
Looks like the style in this file is to have the opening curly brace on a separate line.  It would be good to match the existing style & move your curly brace down onto the next line (unless the curly brace has to stay on the same line to compile correctly)

Reply:
Thanks! Actually this callback is deleted in the updated commit of the PR. Have fixed these styling issues.